vcd/resource_vcd_catalog.go

```diff
@@ -86,6 +87,13 @@ func resourceVcdCatalog() *schema.Resource {
 				Sensitive:   true,
 				Description: "An optional password to access the catalog. Only ASCII characters are allowed in a valid password.",
 			},
+			"metadata": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Key and value pairs for catalog metadata",
+				// For now underlying go-vcloud-director repo only supports
+				// a value of type String in this map.
```

Comment:
Hm. Do you mean that we can't set a number?

More than here, should we state that this is strings-only in the [website/docs/r/catalog.html.markdown](https://github.com/vmware/terraform-provider-vcd/pull/780/files#diff-bf2145122feda798ffa55bd1e73beda9a95863a60aada75b248fcf9e9cc5b327) ?

CC: @vbauzysvmware

Reply:
Let me give some light in this thing.

As per the PR that I have opened in go-vcloud SDK vmware/go-vcloud-director#430 , now metadata methods allow passing the metadata typeValue. Before this, it was hardcoded the stringType. Now it can be anything. (from four types, only stringType and numberType have been tested. DateType and booleanType are not. This thing is documented in the code). Please check https://github.com/mikeletux/go-vcloud-director/blob/f134041848100263630e762d599f4a37457b3b14/govcd/metadata.go#L124-L127

The thing is, as per Terraform schema limitations, we don't want to have additional depth in the schema itself. That's why we keep a map and all metadata entries are enforced to Strings, instead of allowing the user choose the type.

Reply:
Let's from terraform perspective use only String. If there will be requirements for other types when we start brainstorming how to solve this. IMHO

Reply:
@mikeletux

Reply:
I don't think is really needed, since the only option so far from Terraform side of things is strings. I'd leave it like that as @vbauzysvmware says and if in the future there is any need, we can think about it.

Reply:
This was discussed a lot, but I vote for leaving strings only in Terraform (because we have a `TypeMap`, to accommodate types, we'd have to create blocks of structures accepting type and value for each record). IMO that is less convenient than a map unless it is really needed.

But I suggest to remove the comment above (here and in data source), because govcd does support other types now and the comment itself is misleading